beacon-send.html

```diff
@@ -40,89 +40,86 @@
 @status beta
 @homepage https://github.com/ebidel/beacon-send
 -->
-<polymer-element name="beacon-send" attributes="data url dontSendOnClose" hidden>
+
 <script>
-(function() {
-
-var UNLOAD_EVENT_ = 'unload';
-
-Polymer({
-
-  /**
-   * Fired when the message was queued. You should only need this event when
-   * using `send()` explicitly.
-   *
-   * @event beacon-queue
-   * @param {Object} detail
-   *   @param {string} detail.success True if the message was queued successfully.
-   */
-
-  /**
-   * The url of the remote server to send data to.
-   *
-   * @attribute url
-   * @type string
-   * @default null
-   */
-  url: null,
-
-  /**
-   * The data to send to the remove server.
-   *
-   * @attribute data
-   * @type ArrayBufferView|Blob|String|FormData
-   * @default null
-   */
-  data: null,
-
-  /**
-   * If `true`, does not send data when the page's `unload` event is fired.
-   *
-   * @attribute dontSendOnClose
-   * @type bool
-   * @default true
-   */
-  dontSendOnClose: false,
-
-  created: function() {
-    if (!('sendBeacon' in navigator)) {
-      console.warn(this.localName +
-                   ': the Beacon API is not available in this browser.');
-    }
-  },
-
-  ready: function() {
-    this.unloadHandler_ = this.send.bind(this);
-    this.dontSendOnCloseChanged();
-  },
-
-  dontSendOnCloseChanged: function() {
-    if (this.dontSendOnClose) {
-      window.removeEventListener(UNLOAD_EVENT_, this.unloadHandler_);
-    } else {
-      window.addEventListener(UNLOAD_EVENT_, this.unloadHandler_);
+  Polymer({
+    is: 'beacon-send',
+
+    /**
+     * Fired when the message was queued. You should only need this event when
+     * using `send()` explicitly.
+     *
+     * @event beacon-queue
+     * @param {Object} detail
+     *   @param {string} detail.success True if the message was queued successfully.
+     */
+
+    properties: {
+
+      /**
+       * The url of the remote server to send data to.
+       *
+       * @attribute url
+       * @type string
+       * @default null
+       */
+      url: String,
+
+      /**
+       * The data to send to the remove server.
+       *
+       * @attribute data
+       * @type ArrayBufferView|Blob|String|FormData
+       * @default null
+       */
+      data: Object,
+
+      /**
+       * If `true`, the data are sent only when explicitly calling send() method.
+       *
+       * @attribute manual
+       * @type bool
+       * @default true
+       */
+      manual: {
+        type: Boolean,
+        value: false,
+        observer: '_manualChanged'
+      },
+    },
+
+    created: function() {
+      if (!('sendBeacon' in navigator)) {
+        console.warn(this.localName + ': the Beacon API is not available in this browser.');
+      }
+    },
+
+    ready: function() {
+      this._unloadHandler = this.send.bind(this);
+      this._manualChanged();
```

Comment:
interesting. if I remove it, the test with `window.dispatchEvent` fails. it looks like the observer isn't called on ready. is that expected:

![screen shot 2015-06-17 at 12 27 29 am](https://cloud.githubusercontent.com/assets/25405/8196887/f02adb08-1488-11e5-889d-d09b72e4a0e7.png)